test/unit/deferred.js

```diff
@@ -880,3 +875,27 @@ QUnit.test( "jQuery.when - chaining", function( assert ) {
 
 	defer.resolve( "other deferred" );
 } );
+
+QUnit.test( "jQuery.when - solitary thenables", function( assert ) {
+
+	assert.expect( 1 );
+
+	var rejected = new Promise( function( resolve, reject ) {
+		setTimeout( function() {
+			reject( "rejected" );
+		}, 100 );
+	} );
+
+	QUnit.stop();
+
+	jQuery.when( rejected ).then(
+		function() {
+			assert.ok( false, "Rejected, solitary, non-Deferred thenable should not resolve" );
+		},
+		function() {
+			assert.ok( true, "Rejected, solitary, non-Deferred thenable rejected properly" );
+		}
+	).always( function() {
```

Comment:
`.always` is a nonstandard method specific to jQuery Deferreds, and not under test in this block. Using `.then` means that it won't need to change if we alter the implementation of `jQuery.when` in the future (a minor benefit to be sure, but one that costs basically nothing now).